app/scenes/channel_info/channel_info.js

```diff
@@ -32,23 +35,67 @@ const style = StyleSheet.create({
     }
 });
 
-export default class ChannelInfo extends PureComponent {
+class ChannelInfo extends PureComponent {
     static propTypes = {
+        intl: intlShape.isRequired,
         currentChannel: PropTypes.object.isRequired,
         currentChannelCreatorName: PropTypes.string,
         currentChannelMemberCount: PropTypes.number,
         isFavorite: PropTypes.bool.isRequired,
+        leaveChannelRequest: PropTypes.object.isRequired,
         theme: PropTypes.object.isRequired,
         actions: PropTypes.shape({
             getChannelStats: PropTypes.func.isRequired,
-            goToChannelMembers: PropTypes.func.isRequired
+            goToChannelMembers: PropTypes.func.isRequired,
+            selectInitialChannel: PropTypes.func.isRequired,
+            goBack: PropTypes.func.isRequired,
+            leaveChannel: PropTypes.func.isRequired
         })
     }
 
     componentDidMount() {
         this.props.actions.getChannelStats(this.props.currentChannel.team_id, this.props.currentChannel.id);
     }
 
+    componentWillReceiveProps(nextProps) {
+        const {leaveChannelRequest, actions} = nextProps;
+        if (
+            leaveChannelRequest !== this.props.leaveChannelRequest &&
+            leaveChannelRequest.status === 'success'
+        ) {
+            actions.goBack();
+            actions.selectInitialChannel(this.props.currentChannel.team_id);
```

Comment:
My worry is that the action for leave channel is this one

```javascript
export function leaveChannel(channel) {
    return async (dispatch, getState) => {
        const {currentId: teamId} = getState().entities.teams;
        serviceLeaveChannel(teamId, channel.id)(dispatch, getState).then(() => {
            if (channel.isCurrent) {
                selectInitialChannel(teamId)(dispatch, getState);
            }
        });
    };
}
````

as you can see it will to the selection of the initial channel, so either remove it from the action and handle that wherever that action is being called or handle this in another way here.